manifest: Updated Matter revision to pull factory data fix (#14743)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

The following list summarizes the most important changes inherited from the upstream Matter:

* Updated:

   * The scripts for factory data generation and related :doc:`matter:nrfconnect_factory_data_configuration` documentation page.

>      Now it is possible to generate both .json and .hex files including factory data using a single script.
>      Previously it was required to do that in two steps and using two separate scripts.
>      The previous method is also supported to keep backward compatibility.

Reviewer: ```suggestion
     Now, you can use a single script to generate both JSON and HEX files that include the factory data using a single script.
     Previously, you would have to do that in two steps using two separate scripts.
     The older method is still supported for backward compatibility.
```
